src/main/java/com/gw2auth/oauth2/server/service/OAuth2Scope.java

```diff
@@ -0,0 +1,19 @@
+package com.gw2auth.oauth2.server.service;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+import java.util.stream.Stream;
+
+public interface OAuth2Scope {
+
+    @JsonValue
+    String oauth2();
+    Stream<OAuth2Scope> effective();
+    boolean isSupported(OAuth2ClientApiVersion version);
+
+    @JsonCreator
+    static OAuth2Scope fromJson(String value) {
+        return OAuth2Scopes.fromOAuth2(value).orElseThrow();
```

Comment:
Nit: Use `orElseThrow(IllegalArgumentException::new)` to properly indicate the source of the failure.